packages/gateway/src/features/transformation/index.ts

```diff
@@ -9,35 +9,75 @@
  * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
  * graphic logo are (registered/a) trademark(s) of Plan International.
  */
+import {
+  GQLBirthRegistrationInput,
+  GQLDeathRegistrationInput,
+  GQLMarriageRegistrationInput
+} from '@gateway/graphql/schema'
 import { IAuthHeader } from '@opencrvs/commons'
+import { Bundle } from '@opencrvs/commons/types'
+import { EVENT_TYPE } from '@gateway/features/fhir/constants'
 
-export type IFieldBuilderFunction = (
-  accumulatedObj: any,
-  fieldValue: string | number | boolean,
-  context: any
+export type Context<A extends string | number | symbol = never> = {
+  authHeader: IAuthHeader
+  event: EVENT_TYPE
+  _index: { [Key in A]: number }
+}
+export type IFieldBuilderFunction<
+  Key extends string | number | symbol,
+  FieldType
+> = (
+  accumulatedObj: Bundle,
+  fieldValue: NonNullable<FieldType>,
+  context: Context<Key>
 ) => void
 
-export interface IFieldBuilders {
-  [key: string]: IFieldBuilderFunction | IFieldBuilders
+type AllInputs = GQLBirthRegistrationInput &
+  GQLDeathRegistrationInput &
+  GQLMarriageRegistrationInput
+
+type IfAny<T, Y, N> = 0 extends 1 & T ? Y : N
+
+export type IFieldBuilders<
```

Comment:
This is what enables fhir-builders to get real types

Reply:
:exploding_head: